pkg/apis/v1alpha5/suite_test.go

```diff
@@ -521,3 +524,65 @@ var _ = Describe("Limits", func() {
 		Expect(provisioner.Spec.Limits.ExceededBy(provisioner.Status.Resources)).To(MatchError("cpu resource usage of 17 exceeds limit of 16"))
 	})
 })
+
+var _ = Describe("Provisioner Annotation", func() {
+	var provisioner *Provisioner
+	BeforeEach(func() {
+		provisioner = &Provisioner{
+			ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+			Spec: ProvisionerSpec{
+				Taints: []v1.Taint{
+					{
+						Key:    "key",
+						Effect: v1.TaintEffectNoExecute,
+					},
+				},
+				StartupTaints: []v1.Taint{
+					{
+						Key:    "key",
+						Effect: v1.TaintEffectNoExecute,
+					},
+				},
+				Labels: map[string]string{
+					"keyLabel": "valueLabel",
+				},
+				KubeletConfiguration: &KubeletConfiguration{
+					MaxPods: ptr.Int32(10),
+				},
+				Annotations: map[string]string{
+					"keyAnnotation": "valueAnnotation",
+				},
+			},
+		}
+	})
+	It("should only change hash when static fields are updated", func() {
```

Comment:
This doesn't seem to test what the test says it tests. This validates that the hash doesn't change when you make a change to a behavior field but it doesn't seem to test other things like:
1. Testing that the hash _does_ change when there is a change to a static drift field
2. Testing that the hash is the same for two Proivioners with different names/metadata but the same spec details